### PR TITLE
Make LocalAddresss a newtype

### PR DESCRIPTION
**Pull request by coot**
It is important in `cardano-node` as it is using it instance heads.

### CODE REVIEW

File: ouroboros-network-framework/src/Ouroboros/Network/Snocket.hs [hunk 1]
```diff
@@ -378,31 +357,58 @@ namedPipeSnocket iocp name = Snocket {
...
+      return sd
 #endif
 
 localAddressFromPath :: FilePath -> LocalAddress
 #if defined(mingw32_HOST_OS)
-localAddressFromPath = id
+localAddressFromPath = LocalAddress
 #else
-localAddressFromPath = Socket.SockAddrUnix
+localAddressFromPath = LocalAddress
```
Comment (karknu) [inline ouroboros-network-framework/src/Ouroboros/Network/Snocket.hs]: Unix and windows case look identical with this change, do you still need `#if`?

Comment (dcoutts) [reply ouroboros-network-framework/src/Ouroboros/Network/Snocket.hs]: Indeed, looks like the `#if` can go.

Comment (coot) [reply ouroboros-network-framework/src/Ouroboros/Network/Snocket.hs]: I'll remove it

**Pull request merged**